service/history/hsm/sm.go

```diff
@@ -34,7 +34,7 @@ var ErrInvalidTransition = errors.New("invalid transition")
 // A TaskRegenerator is invoked to regenerate tasks post state-based replication or when refreshing all tasks for a
 // workflow.
 type TaskRegenerator interface {
-	RegenerateTasks() ([]Task, error)
+	RegenerateTasks(*Node) ([]Task, error)
```

Comment:
this is for allowing the implementation to have access to child state machines?

Reply:
Or parents, probably more for parents but could apply to both.